x-pack/legacy/plugins/apm/public/components/app/Settings/AddSettings/AddSettingFlyout.tsx

```diff
@@ -0,0 +1,344 @@
+            <EuiFlexItem grow={false}>
+              <EuiButton
+                fill
+                isDisabled={
+                  !(
+                    (selectedConfig && isSampleRateValid) ||
+                    (!selectedConfig &&
+                      serviceName &&
+                      environment &&
+                      isSelectedEnvironmentValid &&
+                      isSampleRateValid)
+                  )
+                }
+                onClick={async (event: React.MouseEvent<HTMLButtonElement>) => {
+                  event.preventDefault();
+                  await saveConfig({
+                    environment,
+                    serviceName,
+                    sampleRate,
+                    configurationId: selectedConfig
+                      ? selectedConfig.id
+                      : undefined
+                  });
+                  onSubmit();
+                }}
+              >
+                {i18n.translate(
+                  'xpack.apm.settings.cm.flyOut.saveConfigurationButtonLabel',
+                  {
+                    defaultMessage: 'Save configuration'
+                  }
+                )}
+              </EuiButton>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlyoutFooter>
+      </EuiFlyout>
+    </EuiPortal>
+  );
+}
+async function deleteConfig(selectedConfig: Config) {
+  try {
+    await deleteCMConfiguration(selectedConfig.id);
+    toastNotifications.addSuccess({
+      title: i18n.translate(
+        'xpack.apm.settings.cm.deleteConfigSucceededTitle',
+        {
+          defaultMessage: 'Configuration was deleted'
+        }
+      ),
+      text: (
+        <FormattedMessage
+          id="xpack.apm.settings.cm.deleteConfigSucceededText"
+          defaultMessage="You have successfully deleted a configuration for {serviceName}. It will take some time to propagate to the agents."
+          values={{
+            serviceName: `"${selectedConfig.service.name}"`
+          }}
+        />
+      )
+    });
+  } catch (error) {
+    toastNotifications.addDanger({
+      title: i18n.translate('xpack.apm.settings.cm.deleteConfigFailedTitle', {
+        defaultMessage: 'Configuration could not be deleted'
+      }),
+      text: (
+        <FormattedMessage
+          id="xpack.apm.settings.cm.deleteConfigFailedText"
+          defaultMessage="Something went wrong when deleting a configuration for {serviceName}. Error: {errorMessage}"
+          values={{
+            serviceName: `"${selectedConfig.service.name}"`,
+            errorMessage: `"${error.message}"`
+          }}
+        />
+      )
+    });
+  }
+}
+
+async function saveConfig({
+  sampleRate,
+  serviceName,
+  environment,
+  configurationId
+}: {
+  sampleRate: number;
+  serviceName: string | undefined;
+  environment: string | undefined;
+  configurationId?: string;
+}) {
+  try {
+    if (isNaN(sampleRate) || !serviceName) {
+      throw new Error('Missing arguments');
```

Comment:
We might want to make this a little more user friendly. But that's probably a nice-to-have.

Reply:
If this error gets thrown, then something is really wrong with the UI. with all the validation checks and disabling the save button, this should never really happen, unless someone is tinkering with the client code.

Reply:
Ah okay. Then maybe just remove the check. Or is it just there to please Typescript?